File: app/src/main/java/uk/co/ourfriendirony/medianotifier/db/movie/MovieDatabaseDefinition.java
package uk.co.ourfriendirony.medianotifier.db.movie;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;
import android.util.Log;

public class MovieDatabaseDefinition extends SQLiteOpenHelper {
    public static final String WATCHED_FALSE = "0";
    public static final String WATCHED_TRUE = "1";

    public static final String ID = "id";
    public static final String SUBID = "sub_id";
    public static final String TITLE = "title";
    public static final String SUBTITLE = "subtitle";
    public static final String DESCRIPTION = "description";
    public static final String RELEASE_DATE = "release_date";
    public static final String EXTERNAL_URL = "external_url";
    public static final String WATCHED = "watched";

    private static final int DATABASE_VERSION = 6;
    public static final String DATABASE_NAME = "movies";
    public static final String TABLE_MOVIES = "movies";

    MovieDatabaseDefinition(Context context) {
        super(context, DATABASE_NAME, null, DATABASE_VERSION);
    }

    @Override
    public void onCreate(SQLiteDatabase db) {
        Log.d("[DB CREATE]", String.valueOf(this.getClass()));
        db.execSQL("CREATE TABLE " + TABLE_MOVIES + " (" +
                ID + " TEXT, " +
                SUBID + " TEXT, " +
                TITLE + " TEXT, " +
                SUBTITLE + " TEXT, " +
                DESCRIPTION + " TEXT, " +
                RELEASE_DATE + " TEXT, " +
                EXTERNAL_URL + " TEXT, " +
                WATCHED + " INTEGER DEFAULT " + WATCHED_FALSE + ", " +
                "PRIMARY KEY (" + ID + ")" +
                ")");
    }

    @Override
    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
        Log.d("[DB UPGRADE]", String.valueOf(this.getClass()) + " version: " + oldVersion + " -> " + newVersion);
        db.execSQL("DROP TABLE IF EXISTS " + TABLE_MOVIES + ";");
        onCreate(db);
    }
}

File: app/src/main/java/uk/co/ourfriendirony/medianotifier/db/movie/MovieDatabaseDefinition.kt
package uk.co.ourfriendirony.medianotifier.db.movie

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper
import android.util.Log
import uk.co.ourfriendirony.medianotifier.general.Constants

class MovieDatabaseDefinition internal constructor(context: Context?) : SQLiteOpenHelper(context, DATABASE_NAME, null, DATABASE_VERSION) {
    override fun onCreate(db: SQLiteDatabase) {
        Log.d("[DB CREATE]", this.javaClass.toString())
        db.execSQL("CREATE TABLE " + TABLE_MOVIES + " (" +
                ID + " TEXT, " +
                SUBID + " TEXT, " +
                TITLE + " TEXT, " +
                SUBTITLE + " TEXT, " +
                DESCRIPTION + " TEXT, " +
                RELEASE_DATE + " TEXT, " +
                EXTERNAL_URL + " TEXT, " +
                PLAYED + " INTEGER DEFAULT " + Constants.DB_FALSE + ", " +
                "PRIMARY KEY (" + ID + ")" +
                ")")
    }

    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
        Log.d("[DB UPGRADE]", this.javaClass.toString() + " version: " + oldVersion + " -> " + newVersion)
        db.execSQL("DROP TABLE IF EXISTS " + TABLE_MOVIES + ";")
        onCreate(db)
    }

    companion object {
        const val ID = "id"
        const val SUBID = "sub_id"
        const val TITLE = "title"
        const val SUBTITLE = "subtitle"
        const val DESCRIPTION = "description"
        const val RELEASE_DATE = "release_date"
        const val EXTERNAL_URL = "external_url"
        const val PLAYED = "watched"
        const val DATABASE_NAME = "movies"
        const val TABLE_MOVIES = "movies"
        private const val DATABASE_VERSION = 6
    }
}